### PR TITLE
Fixing multiple issues in merged/HPMT pipeline

### LINKED ISSUE #82: Fix `EPC_AVAILABLE`/`MCS_AVAILABLE` flags in merge_proc_datasets.py
It's not correct after I change I did to UPRN

### LINKED ISSUE #88: Remove "Non-domestic" installations from HPMT dataset

### LINKED ISSUE #89: Add `PHOTO_SUPPLY` back to HPMT dataset

### LINKED ISSUE #92: Multiple MCS records associated to the same EPC record in HPMT dataset

### LINKED ISSUE #93: Joining to installers leading to duplicate records in HPMT dataset

### LINKED ISSUE #94: Installers postcode wrongly renamed as installation postcode (HPMT dataset)

### LINKED ISSUE #95: Removing commercial/non domestic installations before merge leading to duplicates

### PR DESCRIPTION
**Pull request by sofiapinto**
# Description 

This PR fixes:
- #88: we remove all types of non-domestic installations now (commercial, non domestic and unspecified when not mapped to an EPC record)
- #95: moved this to after EPC and MCS are concatenated, as doing it before was leading to duplicated records
- #89: as requested by Luca
- #92: in `asf_core_data/pipeline/data_joining/install_date_computation.py` when we were getting the UPRN mapping, multiple installations where mapped to the same UPRN. This might be happening because a home gets multiple HPs over time (possible but unlikely at this point) or because multiple homes with installations in similar addresses (likely in same building) are being mapped to the same UPRN. For those we now remove the mapping with EPC and treat them as if they're not in EPC (as we can't be sure which one is the correct mapping)
- #93 : we were getting duplicates because of the way we were merging to epc/mcs to installers data. This is now fixed;
- #94 : installers postcode was being renamed to POSTCODE which is the installation postcode. This is now fixed.

closes #95
closes #94 
closes #93 
closes #92 
closes #89 
closes #88 

# Instructions for Reviewer(s)

## Review

Hey @Jack-Vines and @sqr00t , thanks for reviewing these scripts. Could one of you please double check if the logic looks okay for the changes in:

-  `asf_core_data/pipeline/data_joining/merge_proc_dataset.py`
- `asf_core_data/pipeline/data_joining/install_date_computation.py`

## Setup
In case you want to run the pipeline to make sure it works:

- clone this repo: `git clone git@github.com:nestauk/asf_core_data.git`
- checkout to the correct branch: `git checkout 88_remove_non_domestic_installations`
- Run `make install`;
- Run `direnv allow`;
- Activate the conda enviroment: `conda activate asf_core_data`;
- Set your API credentials as environment variable, i.e run `export COMPANIES_HOUSE_API_KEY="ADD_YOUR_API_KEY_HERE"` in the command line and replace `ADD_YOUR_API_KEY_HERE` with your API key credentials.
- Run `python asf_core_data/pipeline/data_joining/merge_proc_dataset.py`
---

### CODE REVIEW

File: asf_core_data/pipeline/data_joining/install_date_computation.py [hunk 1]
```diff
@@ -117,8 +126,6 @@ def compute_hp_install_date(
     mcs_hp_date_dict = get_mcs_install_date_mapping()
     df["HP_INSTALL_DATE"] = df["UPRN"].map(mcs_hp_date_dict)
 
-    df = df[df["INSPECTION_DATE"].notna()]
```
Comment (sqr00t) [inline asf_core_data/pipeline/data_joining/install_date_computation.py:120]: Very happy you've moved this into edge cases handling "separately".

**Review by Jack-Vines — APPROVED** [review]
Thanks @sofiapinto, all looks good to me!

File: asf_core_data/pipeline/data_joining/merge_proc_datasets.py [hunk 2]
```diff
@@ -178,24 +309,35 @@ def add_mcs_installer_data(
...
     newest_hist_inst_batch = data_batches.get_latest_hist_installers()
 
-    # Load MCS
+    # Load MCS installer data
     mcs_instllr_data = data_getters.load_s3_data(
         base_config.BUCKET_NAME, newest_hist_inst_batch, usecols=usecols
     )
 
-    mcs_instllr_data.rename(columns={"postcode": "POSTCODE"}, inplace=True)
+    mcs_instllr_data.rename(columns={"postcode": "installer_postcode"}, inplace=True)
```
Comment (sqr00t) [inline asf_core_data/pipeline/data_joining/merge_proc_datasets.py:319]: Good clarification of column name!

File: asf_core_data/pipeline/data_joining/merge_proc_datasets.py [hunk 3]
```diff
@@ -178,24 +309,35 @@ def add_mcs_installer_data(
...
+    df["installation_company_mcs_number"] = df["installation_company_mcs_number"].apply(
+        lambda x: x.split(" ")[1] if not pd.isnull(x) else x
+    )
 
     merged_df = df.merge(
         right=mcs_instllr_data,
         how="outer",
-        left_on=["company_unique_id", "installer_name"],
-        right_on=["company_unique_id", "company_name"],
+        left_on=["company_unique_id", "installation_company_mcs_number"],
```
Comment (sqr00t) [inline asf_core_data/pipeline/data_joining/merge_proc_datasets.py:334]: This makes more sense than previously!

**Review by sqr00t — APPROVED** [review]

**Comment by sqr00t** [comment]
Lgtm too! Thanks @sofiapinto

**Comment by sofiapinto** [comment]
Thanks a lot for taking a look at the PR @sqr00t & @Jack-Vines :D most up to date version of HPMT is now on asf-core-data's S3 bucket.

**Pull request merged**